Inc/HALAL/Services/Communication/Ethernet/TCP/ServerSocket.hpp

```diff
@@ -45,10 +48,34 @@ class ServerSocket : public OrderProtocol{
 
 	void operator=(ServerSocket&& other);
 
+	/**
+	* @brief ends the connection between the server and the client. 
+	*/
 	void close();
 
+	/**
+	* @brief process the data received by the client orders. It is meant to be called only by Lwip on the receive_callback
+	*
+	* reads all the data received by the server in the ethernet buffer, packet by packet.
+	* Then, for each packet, it processes it depending on the order id (default behavior is not process) and removes it from the buffer. 
+	* This makes so the receive_callback (and thus the Socket) can only process declared orders, and ignores all other packets. 
+	*/
 	void process_data();
 
+	/**
+	 * @brief saves the order data into the tx_packet_buffer so it can be sent when a connection is accepted
+	 *
+	 * @param order the order to send, which contains the data and id of the message
+	 * @return true if the data could be allocated in the buffer, false otherwise
+	 */
+	bool queue_order(Order& order);
```

Comment:
I don't like the naming of this, I understand where it is coming from, "queuing" as a verb, but I believe something like:
_add_order_to_queue()_ is more clear and easier to grasp